### PR TITLE
Override typing_extensions package

### PR DESCRIPTION
**Pull request by christophfroehlich**
Fixes broken Debian Builds like [this one](https://github.com/ros-controls/ros2_control/actions/runs/11711763616/job/32620991227?pr=1860)

### CODE REVIEW

File: Dockerfile.debian12 [hunk 1]
```diff
@@ -87,9 +87,10 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
   python3-flake8 \
   python3-coverage \
   && \
-# colcon-devtools>0.2.3 is not released for debian12, so override it from pip
+  # colcon-devtools>0.2.3 is not released for debian12, so override it with pip
+  # typing_extensions needs to be >4.7.0, but is available on apt with 4.4.0 only
```
Comment (saikishor) [inline Dockerfile.debian12:91]: So with this the CI build should be good, but the local build might have the same issue right?

Correct me if I'm wrong :)

**Review by saikishor — APPROVED** [review]
LGTM

File: Dockerfile.debian12 (hunk 1 above)
Comment (christophfroehlich) [reply Dockerfile.debian12:91]: yes, but I see the dockerfile as an example to setup your local debian to get ros2_control framework running ;)

Comment (saikishor) [reply Dockerfile.debian12:91]: Sure!

**Comment by saikishor** [comment]
LGTM

**Pull request merged**